pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -398,6 +398,7 @@ public UnAckedMessageTracker getUnAckedMessageTracker() {
             messageProcessed(message);
             return beforeConsume(message);
         } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
             stats.incrementNumReceiveFailed();
             throw PulsarClientException.unwrap(e);
```

Comment:
@eolivelli I'm not sure why this is nessary, it have thrown PulsarClientException here.

Reply:
BTW, please add same above in the method internalReceive/internalBatchReceive if this is nessary

Reply:
Okay! I'll do it